src/dapps/SecureTransaction/SecureTransaction.vue

```diff
@@ -0,0 +1,460 @@
+          <p>Gwei</p>
+          <i
+            class="fa fa-check-circle good-button not-good"
+            aria-hidden="true"
+          />
+        </div>
+      </div>
+    </div>
+    <div class="submit-button-container">
+      <div>
+        <i
+          :class="[
+            validNetwork ? '' : 'not-good',
+            'fa fa-check-circle good-button'
+          ]"
+          aria-hidden="true"
+        />
+        You must use the {{ chainIDEnglish }} network of your web3 provider
+        (metamask, etc.) to use SafeSend.
+      </div>
+      <br />
+      <div>
+        Your estimated SafeSend Fee (in addition to gas):&nbsp;~{{
+          safeSendPriceEstimate
+        }}&nbsp;ETH
+      </div>
+      <br />
+      <div
+        :class="[
+          validAddress && validAmount && validNetwork && address.length !== 0
+            ? ''
+            : 'disabled',
+          'submit-button large-round-button-green-filled'
+        ]"
+        @click="confirmationModalOpen"
+      >
+        Send Secured Transaction
+      </div>
+      <interface-bottom-text
+        :link-text="$t('interface.learnMore')"
+        :question="$t('interface.haveIssues')"
+        link="https://heycoral.com/safesend/index.html"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapGetters } from 'vuex';
+import InterfaceContainerTitle from '@/layouts/InterfaceLayout/components/InterfaceContainerTitle';
+import { CoralConfig } from './config';
+import InterfaceBottomText from '@/components/InterfaceBottomText';
+import Blockie from '@/components/Blockie';
+import normalise from '@/helpers/normalise';
+import BigNumber from 'bignumber.js';
+import * as unit from 'ethjs-unit';
+import fetch from 'node-fetch';
+import utils from 'web3-utils';
+
+export default {
+  components: {
+    'interface-container-title': InterfaceContainerTitle,
+    'interface-bottom-text': InterfaceBottomText,
+    blockie: Blockie
+  },
+  props: {
+    tokensWithBalance: {
+      type: Array,
+      default: function() {
+        return [];
+      }
+    },
+    getBalance: {
+      type: Function,
+      default: function() {}
+    },
+    highestGas: {
+      type: Number,
+      default: 0
+    }
+  },
+  data() {
+    return {
+      protectionLevel: 'low',
+      advancedExpend: false,
+      validAmount: false,
+      validNetwork: false,
+      validAddress: false,
+      chainIDEnglish: CoralConfig.chainIDEnglish || 'mainnet',
+      minimumAmount: 0.01,
+      amount: 0,
+      safeSendPriceEstimate: 0,
+      amountValid: true,
+      nonce: 0,
+      gasLimit: 21000,
+      data: '0x',
+      gasAmount: 0,
+      parsedBalance: 0,
+      address: '',
+      transactionFee: 0,
+      selectedCurrency: { symbol: 'ETH', name: 'Ethereum' },
+      raw: {},
+      signedTx: '',
+      resolvedAddress: ''
+    };
+  },
+  computed: {
+    ...mapGetters({
+      account: 'account',
+      gasPrice: 'gasPrice',
+      web3: 'web3',
+      wallet: 'wallet',
```

Comment:
can you replace the instances of `wallet` usage to `await web3.eth.getCoinbase()` also take a look at our new send Transaction codebase, which is lot cleaner https://github.com/MyEtherWallet/MyEtherWallet/blob/develop/src/layouts/InterfaceLayout/containers/SendCurrencyContainer/SendCurrencyContainer.vue